timrlink.net.CLI/Actions/ProjectTimeDatabaseExportAction.cs

```diff
@@ -57,36 +119,62 @@ public async Task Execute()
-                        StartTime = pt.startTime,
-                        EndTime = pt.endTime,
-                        Duration = pt.duration,
-                        BreakTime = pt.breakTime,
-                        Changed = pt.changed,
-                        Closed = pt.closed,
-                        StartPosition = LatLon(pt.startPosition),
-                        EndPosition = LatLon(pt.endPosition),
-                        LastModifiedTime = pt.lastModifiedTime,
-                        Task = JsonConvert.SerializeObject(BuildTaskPath(pt.taskUuid, taskDict).Select(task => task.name).ToArray()),
-                        Description = pt.description,
-                        Billable = pt.billable
-                    };
+                    var projectTime = context.ProjectTimes
+                        .FirstOrDefault(projectTime => projectTime.UUID.ToString() == pt.uuid) ?? new ProjectTime();
+                    
+                    var start1 = pt.GetStartTimeOffset().DateTime;
+                    var end1 = pt.GetEndTimeOffset().DateTime;
+
+                    projectTime.UUID = Guid.Parse(pt.uuid);
+                    projectTime.User = user != null ? $"{user.lastname} {user.firstname}" : pt.userUuid;
+                    projectTime.StartTime = pt.GetStartTimeOffset().LocalDateTime;
+                    projectTime.EndTime = pt.GetEndTimeOffset().LocalDateTime;
+                    projectTime.Duration = pt.duration;
+                    projectTime.BreakTime = pt.breakTime;
+                    projectTime.Changed = pt.changed;
+                    projectTime.Closed = pt.closed;
+                    projectTime.StartPosition = LatLon(pt.startPosition);
+                    projectTime.EndPosition = LatLon(pt.endPosition);
+                    projectTime.LastModifiedTime = pt.lastModifiedTime;
+                    projectTime.Task =
+                        JsonConvert.SerializeObject(BuildTaskPath(pt.taskUuid, taskDict).Select(task => task.name)
+                            .ToArray());
+                    projectTime.Description = pt.description;
+                    projectTime.Billable = pt.billable;
+                    projectTime.Deleted = null;
+
+                    return projectTime;
                 });
-
-                await context.ProjectTimes.AddOrUpdateRange(dbEntities);
-                await context.SaveChangesAsync();
+                
+                await context.ProjectTimes.AddOrUpdateRange(dbEntities);    
             }
 
-            await context.SetMetadata(new Metadata(Metadata.KEY_LAST_PROJECTTIME_IMPORT, importTime.ToString("o", CultureInfo.InvariantCulture)));
+            if (dateSpan != null)
+            {
+                // Flag records that are not found anymore Deleted.
+                var projectTimesInDatabase = context.ProjectTimes.Where(projectTime =>
+                        projectTime.StartTime >= dateSpan.From && projectTime.EndTime <= dateSpan.To)
```

Comment:
`projectTime.EndTime <= dateSpan.To` won't match when `EndTime` has a time, because `dateSpan.To` is a date only.